src/Agent.Worker/StepsRunner.cs

```diff
@@ -82,10 +82,19 @@ public sealed class StepsRunner : AgentService, IStepsRunner
                 }
                 catch (OperationCanceledException ex)
                 {
-                    // Log the exception and cancel the step.
-                    Trace.Error($"Caught cancellation exception from step: {ex}");
-                    step.ExecutionContext.Error(ex);
-                    step.ExecutionContext.Result = TaskResult.Canceled;
+                    if (step.ExecutionContext.TimeoutExceeded)
```

Comment:
This same type of logic needs to be added below for the OperationCanceledException catch block for the async command.